File: iosApp/iosApp/Features/JobOffer/SortOption.swift
//
//  SortOption.swift
//  iosApp
//
//  Created by DAN BIZWA on 19/09/2026.
//

import SwiftUI

/// Options de tri. La valeur brute est la CLÉ de traduction (`sort_*` dans Localizable.strings), pas un texte.
enum SortOption: String, CaseIterable, Identifiable {
    case dateDesc = "sort_newest"
    case dateAsc = "sort_oldest"
    case alphaAsc = "sort_az"
    case alphaDesc = "sort_za"
    var id: String { rawValue }

    var titleKey: LocalizedStringKey { LocalizedStringKey(rawValue) }
}

File: androidApp/src/main/kotlin/com/dmb/joblog/ui/joboffer/SortOption.kt
package com.dmb.joblog.ui.joboffer

import androidx.annotation.StringRes
import com.dmb.joblog.R

/** Options de tri ; le libellé est une ressource (`sort_*` dans strings.xml), résolue dans la langue courante par l'écran. */
enum class SortOption(@StringRes val labelRes: Int) {
    DATE_DESC(R.string.sort_newest),
    DATE_ASC(R.string.sort_oldest),
    ALPHA_ASC(R.string.sort_az),
    ALPHA_DESC(R.string.sort_za)
}
